rfcs/000 - RFC Template.md

```diff
@@ -0,0 +1,55 @@
+# 000 - RFC Template
+
+## Current Status
+
+### Proposed
+
+YYYY-MM-DD
+
+### Accepted
+
+YYYY-MM-DD
+
+#### Approvers
+
+- Full Name <email@example.com>
+
+### Implementation
+
+- [ ] Implemented: [One or more PRs](https://github.com/iasql/iasql-engine/some-pr-link-here) YYYY-MM-DD
+- [ ] Revoked/Superceded by: [RFC ###](./000 - RFC Template.md) YYYY-MM-DD
+
+## Author(s)
+
+- Author A <a@example.com>
+- Author B <b@example.com>
+
+## Summary
+
+A brief, one to two paragraph summary of the problem and proposed solution goes here. The name of the PR *must* include "RFC" in it to be searchable.
+
+## Proposal
+
+A more detailed description of the proposed changes, what they will solve, and why it should be done. Diagrams and code examples very welcome!
+
+Reviewers should *not* bring up alternatives in this portion unless the template is not being followed by the RFC author (which the RFC author should note in the PR with a detailed reason why). Reviewers should also not let personal distaste for a solution be the driving factor behind criticism of a proposal, there should be some rationale behind a criticism, though you can still voice your distaste since that means there's probably *something* there that perhaps another reviewer could spot (but distate on its own should not block).
+
+Most importantly, be civil on both proposals and reviews. `iasql` is meant to be an approachable tool for developers and if we want to make it better we need to be approachable to each other. Some parts of the language may have been mistakes, but they certainly weren't intentional and all parts were thought over by prior contributors. New proposals come from people who see something that doesn't sit well with them and they have put forth the energy to write a proposal and we should be thankful that they care and want to make it better.
+
+Ideally everyone can come to a refined version of the RFC that satisfies all arguments and is better than what anyone person could have come up with, but if an RFC is divisive, the "winning" side should be gracious, and the "losing" side should hopefully accept that the proposal was contentious.
+
```

Comment:
But I'd change it to just `Semver` instead of `SemBDD` -- since this is not Alan and the idea of baking BDD testing into the language and using that for version change decisions automatically doesn't exist in Typescript.

Reply:
Yes I had my doubts and removed it consciously thinking that with the `Affected Components` was enogh but it is true is not exactly the same and could be important. I'll add it back